src/ring_theory/tensor_product.lean

```diff
@@ -717,3 +717,70 @@ end monoidal
 end tensor_product
 
 end algebra
+
+namespace tensor_product
+
+variables {R M N S: Type*} [comm_semiring R] [add_comm_monoid M] [add_comm_monoid N] [semiring S]
+variables [module R M] [module R N] [algebra R S]
+
+/--
+If `S` is an `R`-algebra, for a pair of morphisms `f : M →ₗ[R] S`, `g : N →ₗ[R] S`,
+We may obtain a map `M ⊗[R] N →ₗ[R] S` via `a ⊗ b ↦ f(a) * g(b)`.
+This may not be that useful in module-sense, but is useful when `M`, `N` are both algebras.
+-/
+def product_map (f : M →ₗ[R] S) (g : N →ₗ[R] S) : M ⊗[R] N →ₗ[R] S :=
+lift {
+  to_fun := λ m, {
+    to_fun := λ n, f m * g n,
+    map_add' := λ n₁ n₂, by rw [g.map_add, mul_add],
+    map_smul' := λ r n, by rw [g.map_smul, mul_smul_comm]
+  },
+  map_add' := λ m₁ m₂, by
+    { ext n, simp only [linear_map.coe_mk, linear_map.add_apply, f.map_add, add_mul] },
+  map_smul' := λ r m, by
+    { ext n, simp only [linear_map.coe_mk, linear_map.smul_apply,
+      f.map_smul, algebra.smul_mul_assoc] }
+}
+
+@[simp] lemma product_map_apply_tmul (f : M →ₗ[R] S) (g : N →ₗ[R] S) (m : M) (n : N) :
+  product_map f g (m ⊗ₜ n) = f m * g n :=
+by apply tensor_product.lift.tmul
+
+end tensor_product
+
+namespace algebra.tensor_product
+
+variables {R A B S: Type*} [comm_semiring R] [semiring A] [semiring B] [comm_semiring S]
+variables [algebra R A] [algebra R B] [algebra R S]
+variables (f : A →ₐ[R] S) (g : B →ₐ[R] S)
+include f g
+
+/--
+If `S` is commutative, for a pair of morphisms `f : A →ₐ[R] S`, `g : B →ₐ[R] S`,
+We obtain a map `A ⊗[R] B →ₐ[R] S` that commutes with `f`, `g` via `a ⊗ b ↦ f(a) * g(b)`.
+-/
+def product_map : A ⊗[R] B →ₐ[R] S :=
+begin
+  apply alg_hom_of_linear_map_tensor_product
+    (tensor_product.product_map f.to_linear_map g.to_linear_map),
+  { intros a₁ a₂ b₁ b₂,
+    unfold alg_hom.to_linear_map,
+    simp only [linear_map.coe_mk, tensor_product.product_map_apply_tmul, alg_hom.map_mul],
+    rw [semiring.mul_assoc, semiring.mul_assoc,
+      ← semiring.mul_assoc (f a₂), ← semiring.mul_assoc (g b₁), mul_comm (f a₂)],
+    refl },
+  { intros r, simp }
+end
```

Comment:
I think the interesting definition here is actually:
```lean
/-- `algebra.lmul'` is an alg_hom on commutative rings-/
def lmul' : S ⊗[R] S →ₐ[R] S :=
alg_hom_of_linear_map_tensor_product (algebra.lmul' R)
  (λ a₁ a₂ b₁ b₂, by simp only [algebra.lmul'_apply, mul_mul_mul_comm])
  (λ r, by simp only [algebra.lmul'_apply, _root_.mul_one])

lemma lmul'_to_linear_map :
  (lmul' R : S ⊗[R] S →ₐ[R] S).to_linear_map = algebra.lmul' R := rfl

@[simp] lemma lmul'_apply (a b : S) : lmul' R (a ⊗ₜ b) = a * b := algebra.lmul'_apply
```
which makes this entire definition just `(lmul' R).comp (map f g)`

Reply:
Your lemmas below can probably then just become about `lmul'`

Reply:
I apparently forgot that it factors through `S ⊗[R] S`.
I've rewrote the definition. Thanks!